casper/src/main/scala/coop/rchain/casper/util/ProtoUtil.scala

```diff
@@ -137,43 +166,52 @@ object ProtoUtil {
     } yield ps.blockNumber).getOrElse(0L)
 
   //Two blocks conflict if they both use the same deploy in different histories
-  def conflicts(b1: BlockMessage,
-                b2: BlockMessage,
-                genesis: BlockMessage,
-                dag: BlockDag,
-                internalMap: Map[BlockHash, BlockMessage]): Boolean = {
-    val gca = DagOperations.greatestCommonAncestor(b1, b2, genesis, dag, internalMap)
-    if (gca == b1 || gca == b2) {
-      //blocks which already exist in each other's chains do not conflict
-      false
-    } else {
-      def getDeploys(b: BlockMessage) =
-        DagOperations
-          .bfTraverse[BlockMessage](Some(b))(parents(_).iterator.map(internalMap(_)))
-          .takeWhile(_ != gca)
-          .flatMap(b => {
-            b.body.map(_.newCode).getOrElse(List.empty[Deploy])
-          })
-          .toSet
-
-      getDeploys(b1).intersect(getDeploys(b2)).nonEmpty
-    }
-  }
-
-  def chooseNonConflicting(blocks: Seq[BlockMessage],
-                           genesis: BlockMessage,
-                           dag: BlockDag,
-                           internalMap: Map[BlockHash, BlockMessage]): Seq[BlockMessage] =
-    blocks
-      .foldLeft(List.empty[BlockMessage]) {
+  def conflicts[F[_]: Monad: BlockStore](b1: BlockMessage,
```

Comment:
We should probably update the logic of this function to make use of the trace logs and say that two blocks don't conflict if they act on disjoint sets of channels. But maybe create a separate ticket for for that instead of doing it as part of this PR.

Reply:
Or I meant see https://rchain.atlassian.net/browse/RHOL-625 for the ticket to do this